inference-engine/src/mkldnn_plugin/nodes/embedding_bag_sum.cpp

```diff
@@ -0,0 +1,203 @@
+        const CNNLayer* layer,
+        size_t required_input_num,
+        size_t indices_idx,
+        size_t per_sample_weights_idx,
+        size_t default_index_idx,
+        const std::set<Precision>& supported_precisions) :
+            INDICES_IDX(indices_idx),
+            PER_SAMPLE_WEIGHTS_IDX(per_sample_weights_idx),
+            DEFAULT_INDEX_IDX(default_index_idx) {
+    try {
+        std::string log_prefix = "Layer EmbeddingBagSum with name '";
+        if (layer->insData.size() < required_input_num || layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << log_prefix << layer->name << "' has incorrect number of input or output edges!";
+        _l_name = layer->name;
+
+        auto inData = layer->insData[0].lock();
+        auto indicesData = layer->insData[INDICES_IDX].lock();
+        if (inData == nullptr || indicesData == nullptr)
+            THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has nullable input data.";
+
+        const auto precision = inData->getTensorDesc().getPrecision();
+        if (!supported_precisions.empty()) {
+            if (supported_precisions.find(precision) == supported_precisions.end())
+                THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has unsupported precision: " << precision.name();
+        } else {
+            static const std::set<Precision> default_supported_precisions =
+                {Precision::FP32, Precision::BF16, Precision::I8, Precision::U8, Precision::I32};
+            if (default_supported_precisions.find(precision) == default_supported_precisions.end())
+                THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has unsupported precision: " << precision.name();
+        }
+
+        if (layer->insData.size() > PER_SAMPLE_WEIGHTS_IDX)
+            _with_weights = true;
+        if (_with_weights) {
+            auto weightsData = layer->insData[PER_SAMPLE_WEIGHTS_IDX].lock();
+            if (weightsData == nullptr)
+                 THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has nullable weights data";
+            if (weightsData->getTensorDesc().getDims() != indicesData->getTensorDesc().getDims())
+                 THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer must have equal shapes for indices and per_sample_weights inputs.";
+        }
+
+        LayerConfig config;
+        config.inConfs.resize(layer->insData.size());
+        for (int i = 0; i < layer->insData.size(); i++) {
+            auto data = layer->insData[i].lock();
+            if (data == nullptr)
+                THROW_IE_EXCEPTION << log_prefix << _l_name << "' layer has nullable input data";
+            config.inConfs[i].desc = TensorDesc(data->getTensorDesc());
+        }
+
+        DataConfig outConfig;
+        outConfig.desc = TensorDesc(inData->getTensorDesc().getPrecision(),
+            layer->outData[0]->getTensorDesc().getDims(),
+            layer->outData[0]->getTensorDesc().getLayout());
```

Comment:
Output layout have to be equal to the input one.

Reply:
FIxed